src/EnergyPlus/ZoneTempPredictorCorrector.cc

```diff
@@ -3416,6 +3421,147 @@ namespace ZoneTempPredictorCorrector {
             }
         }
 
+        // Setpoint revision for onoff thermostat
+        Real64 TempTole = 0.02;
+        Real64 Tprev;
+        if (NumOnOffCtrZone > 0) {
+            for (RelativeZoneNum = 1; RelativeZoneNum <= NumTempControlledZones; ++RelativeZoneNum) {
+                if (TempControlledZone(RelativeZoneNum).DeltaTCutSet > 0.0) {
+                    if (ShortenTimeStepSys) {
+                        TempControlledZone(RelativeZoneNum).HeatModeLast = TempControlledZone(RelativeZoneNum).HeatModeLastSave;
+                        TempControlledZone(RelativeZoneNum).CoolModeLast = TempControlledZone(RelativeZoneNum).CoolModeLastSave;
+                    } else {
+                        TempControlledZone(RelativeZoneNum).HeatModeLastSave = TempControlledZone(RelativeZoneNum).HeatModeLast;
+                        TempControlledZone(RelativeZoneNum).CoolModeLastSave = TempControlledZone(RelativeZoneNum).CoolModeLast;
+                    }
+                    ZoneNum = TempControlledZone(RelativeZoneNum).ActualZoneNum;
+                    {   auto const SELECT_CASE_var(TempControlType(ZoneNum));
+                    TempControlledZone(RelativeZoneNum).CoolOffFlag = false;
+                    TempControlledZone(RelativeZoneNum).HeatOffFlag = false;
+
+                    if (SELECT_CASE_var == SingleHeatingSetPoint) {
+                        TempZoneThermostatSetPoint(ZoneNum) = TempControlledZone(RelativeZoneNum).ZoneThermostatSetPointLo;
+                        ZoneThermostatSetPointLo(ZoneNum) = TempControlledZone(RelativeZoneNum).ZoneThermostatSetPointLo;
+                        if (ZoneAirSolutionAlgo == Use3rdOrder) {
+                            Tprev = MAT(ZoneNum);
+                            if (ShortenTimeStepSys) Tprev = XMPT(ZoneNum);
+                        }
+                        else {
+                            Tprev = ZoneT1(ZoneNum);
+                        }
```

Comment:
This section that sets `Tprev` looks identical for all three thermostat types, so it could appear just once before line 3442 unless I'm missing something.

Reply:
This comment refers to lines 3445:3451